README.markdown

```diff
@@ -7793,10 +7793,38 @@ tcpsock:setoption
 
 **context:** *rewrite_by_lua&#42;, access_by_lua&#42;, content_by_lua&#42;, ngx.timer.&#42;, ssl_certificate_by_lua&#42;, ssl_session_fetch_by_lua&#42;*
 
-This function is added for [LuaSocket](http://w3.impa.br/~diego/software/luasocket/tcp.html) API compatibility and does nothing for now. Its functionality will be implemented in future.
+This function is added for [LuaSocket](http://w3.impa.br/~diego/software/luasocket/tcp.html) API compatibility and does nothing for now. Its functionality is implemented `v0.10.18`.
 
 This feature was first introduced in the `v0.5.0rc1` release.
 
+`option` is a string with the option name, and value depends on the option being set:
+
+* `keepalive`
+	Setting this option to true enables sending of keep-alive messages on
+	connection-oriented sockets(e.g., tcpsock:setoption("keepalive", 1) and
+	tcpsock:setoption("keepalive", false)).
+
+* `reuseaddr`
+	Enable this option indicates that the rules used in validating addresses
+	supplied in a call to bind should allow reuse of local addresses(e.g.,
+	tcpsock:setoption("reuseaddr", true) and tcpsock:setoption("reuseaddr", 0)).
+
+* `tcp-nodelay`
+	Setting this option to true disables the Nagle's algorithm for the connectio(e.g.,
+	tcpsock:setoption("tcp-nodelay", 1) and tcpsock:setoption("tcp-nodelay", false)).
+
+* `sndbuf`
+	Sets the maximum socket send buffer in bytes. The kernel doubles this value
+	(to allow space for bookkeeping overhead) when it is set using setsockopt()
+	(e.g., tcpsock:setoption("sndbuf", 1024 * 10)).
+
+* `rcvbuf`
+	Sets the maximum socket receive buffer in bytes. The kernel doubles this value
+	(to allow space for bookkeeping overhead) when it is set using setsockopt()
+	(e.g., tcpsock:setoption("rcvbuf", 1024 * 10)).
+
```

Comment:
We should add some notes and code examples for avoiding calling `setoption()` repeatedly on connections from the connection pools.

Reply:
accept